spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -418,15 +426,72 @@ def sample_allocation(obj)
         end
 
         it 'include the stack and sample counts for the objects still left alive' do
-          skip('FIXME: Unblock CI -- flaky on Ruby 3.3') if RUBY_VERSION.start_with?('3.3.')
+          # There should be 3 different allocation class labels so we expect 3 different heap samples
+          expect(heap_samples.size).to eq(3)
 
-          # We sample from 2 distinct locations
-          expect(heap_samples.size).to eq(2)
+          string_sample = heap_samples.find { |s| s.labels[:'allocation class'] == 'String' }
+          expect(string_sample.values).to match(
+            hash_including(
+              :"heap-live-samples" => sample_rate,
+            )
+          )
+          expect(string_sample.labels).to match(
+            {
+              :'allocation class' => 'String',
+              :'gc gen age' => be_a(Integer).and(be >= 0),
+            }
+          )
 
-          sum_heap_samples = 0
-          heap_samples.each { |s| sum_heap_samples += s.values[:'heap-live-samples'] }
+          array_sample = heap_samples.find { |s| s.labels[:'allocation class'] == 'Array' }
+          expect(array_sample.values).to match(
+            hash_including(
+              :"heap-live-samples" => sample_rate,
+            )
+          )
+          expect(array_sample.labels).to match(
+            {
+              :'allocation class' => 'Array',
+              :'gc gen age' => be_a(Integer).and(be >= 0),
+            }
+          )
 
-          expect(sum_heap_samples).to(eq([a_string, an_array, a_hash].size * sample_rate))
+          hash_sample = heap_samples.find { |s| s.labels[:'allocation class'] == 'Hash' }
+          expect(hash_sample.values).to match(
+            hash_including(
+              :"heap-live-samples" => sample_rate,
+            )
+          )
+          expect(hash_sample.labels).to match(
+            {
+              :'allocation class' => 'Hash',
+              :'gc gen age' => be_a(Integer).and(be >= 0),
+            }
+          )
+        end
```

Comment:
Minor: Since the checks are pretty much the same, they can be simplified a bit:

```ruby
          expect(heap_samples.map { |it| it.labels[:'allocation class'] }).to include('String', 'Array', 'Hash')
          expect(heap_samples.map(&:labels)).to all(match(:'gc gen age' => be_a(Integer).and(be >= 0)))
```

Reply:
Ah yes failed to spot this after splitting the tests 😅